agent/statemanager/state_manager.go

```diff
@@ -192,6 +194,7 @@ func AddSaveable(name string, saveable Saveable) Option {
 // Save triggers a save to file, though respects a minimum save interval to wait
 // between saves.
 func (manager *basicStateManager) Save() error {
+	defer metrics.MetricsEngineGlobal.RecordStateManagerMetric("SAVE")()
```

Comment:
should this be in ForceSave instead? sometimes we don't actually save here because we have a minimum save interval, and in that case i don't think that's the time we want to track

Reply:
The Save() function determines if ForceSave() should be called, we are only tracking the times when Save() is called (regardless of whether a save to disk was made). What are your thoughts on this? We could alternatively produce a metric for the ForceSave() call since all writes to disk go through this method.